devdoc/rc_string_requirements.md

```diff
@@ -72,6 +72,26 @@ MOCKABLE_FUNCTION(, THANDLE(RC_STRING), rc_string_create, const char*, string);
 
 **SRS_RC_STRING_01_006: [** If any error occurs, `rc_string_create` shall fail and return `NULL`. **]**
 
+## rc_string_create_with_format
+
+```c
+THANDLE(RC_STRING) rc_string_create_with_format(const char* format, ...);
+```
+
+`rc_string_create_with_format` creates a new ref counted string by using the format convention as in sprintf.
+
+If `format` is `NULL`, `rc_string_create_with_format` shall fail and return `NULL`. 
+
+Otherwise, `rc_string_create_with_format` shall determine the total number of characters written using the variable number of arguments.  
+
+`rc_string_create_with_format` shall allocate memory for the `THANDLE(RC_STRING)`.
```

Comment:
>allocate memory for the `THANDLE(RC_STRING)`. [](http://example.com/codeflow?start=37&length=45)

I think we should specify here that we're allocating memory for THANDLE(RC_STRING) and the number of bytes needed for the resulting formatted string.